linera-execution/src/system.rs

```diff
@@ -638,14 +646,31 @@ where
         self.debit(&source, amount).await?;
         match recipient {
             Recipient::Account(account) => {
-                let message = SystemMessage::Credit {
-                    amount,
-                    source,
-                    target: account.owner,
-                };
-                Ok(Some(
-                    OutgoingMessage::new(account.chain_id, message).with_kind(MessageKind::Tracked),
-                ))
+                // Check if destination is on the same chain
```

Comment:
nit: This comment would be barely needed if we used a more specific name for `chain_id`, like `local_chain_id`, `source_chain_id`, `this_chain_id`.

Reply:
We could also have `assert_eq!(chain_id, self.context().extra().chain_id);`

Reply:
Ok.